scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-formal-definition.js

```diff
@@ -0,0 +1,58 @@
+const { select } = require("hast-util-select");
+
+const { isMacro, sliceSection, findExtraneousNode } = require("./utils");
+
+function handleDataFormalDefinition(tree, logger) {
+  const id = "Formal_definition";
+  const body = select(`body`, tree);
+  const heading = select(`h2#${id}`, body);
+
+  if (heading === null) {
+    logger.expected(body, `h2#${id}`, "expected-heading");
+    return null;
+  }
+
+  // Section must have an immediate child P element containing the CSSInfo macro
+  const section = sliceSection(heading, body);
```

Comment:
Yeah, this is the right thing to do. Though to do it kinda gracefully ended up being a slightly bigger change than I anticipated. It might help to look at that commit alone. If you want me to move that change to its own PR, let me know.